app/src/sharedTest/java/org/oppia/android/app/home/HomeFragmentViewModelsTest.kt

```diff
@@ -0,0 +1,1233 @@
+import org.oppia.android.app.shim.IntentFactoryShimModule
+import org.oppia.android.app.shim.ViewBindingShimModule
+import org.oppia.android.app.testing.HomeFragmentTestActivity
+import org.oppia.android.app.testing.HomeFragmentTestActivity.Companion.createHomeFragmentTestActivity
+import org.oppia.android.domain.classify.InteractionsModule
+import org.oppia.android.domain.classify.rules.continueinteraction.ContinueModule
+import org.oppia.android.domain.classify.rules.dragAndDropSortInput.DragDropSortInputModule
+import org.oppia.android.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.android.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.android.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.android.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.oppia.android.util.system.OppiaClock
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// Time: Wed Apr 24 2019 08:22:00
+private const val MORNING_TIMESTAMP = 1556094120000
+
+// Time: Tue Apr 23 2019 23:22:00
+private const val EVENING_TIMESTAMP = 1556061720000
+
+/** Tests for [HomeViewModel]s data. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = HomeFragmentViewModelsTest.TestApplication::class,
+  manifest = Config.NONE
+)
+class HomeFragmentViewModelsTest {
```

Comment:
2 things:

1. Please split this up such that you have one test file for each view model
2. Please also add tests for hash code (e.g. tests that cover the needed behaviors in https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object) and https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#hashCode())
3. Expand test coverage to match the needed behaviors above

Reply:
Done

For hash I check that object equals => hash codes are same, and that hash code does not change during application. Is there another test case you are thinking of? Seems redundant to test each hashcode equality for each of the equals() cases

Reply:
This plan sounds good--thanks!